tests/integration_tests/modules/test_ubuntu_advantage.py

```diff
@@ -31,6 +31,8 @@
 package_upgrade: true
 packages:
 - ubuntu-advantage-tools
+bootcmd:
+- systemctl mask ua-auto-attach.service
```

Comment:
`ua-auto-attach.service` will check for UA cloud-init data and noop
`ubuntu-advantage.service` is the GCE daemon that will attach anyway - I think that is the one you need to mask for this test to work

Reply:
Now that I wrote that, `ubuntu-advantage.service` is `After: cloud-config.service` so it shouldn't be attaching before cloud-init would. 

It will automatically re-auto-attach if you run `ua detach` on GCE though

Reply:
I have disabled both units and performed a `pro detach` to be sure it is really detached on second boot. Please, @orndorffgrant, validate this is a correct set-up on first boot.